packages/dapp/src/components/Navbar/NavbarManagement.tsx

```diff
@@ -1,43 +1,67 @@
 import { Chain } from '@lifi/types';
 import MenuIcon from '@mui/icons-material/Menu';
 import { Typography } from '@mui/material';
-import { WalletManagementButtons } from '@transferto/shared/src';
-import { ThemeSwitch } from '@transferto/shared/src/atoms/ThemeSwitch';
-import { useSettings } from '@transferto/shared/src/hooks';
-import { useEffect, useMemo, useRef } from 'react';
+import {
+  MenuContextProps,
+  SettingsContextProps,
+  WalletManagementButtons,
+} from '@transferto/shared/src';
+import { useEffect, useLayoutEffect, useMemo, useRef } from 'react';
 import { useTranslation } from 'react-i18next';
-import { SubMenuKeys } from '../../const';
 import { useChainInfos } from '../../providers/ChainInfosProvider';
-import { useMenu } from '../../providers/MenuProvider';
 import { useWallet } from '../../providers/WalletProvider';
+import { useSettingsStore } from '../../stores';
+import { useMenuStore } from '../../stores/menu';
+import { ThemeSwitch } from '../ThemeSwitch';
 import {
   NavbarDropdownButton,
   NavbarManagement as NavbarManagementContainer,
 } from './Navbar.style';
 
 const NavbarManagement = () => {
-  const anchorRef = useRef<HTMLButtonElement>(null);
-  const settings = useSettings();
-  const menu = useMenu();
+  const anchorRef = useRef<any>(null);
+  const onWalletDisconnect = useSettingsStore(
+    (state: SettingsContextProps) => state.onWalletDisconnect,
+  );
+
+  const onOpenNavbarMainMenu = useMenuStore(
+    (state: MenuContextProps) => state.onOpenNavbarMainMenu,
+  );
+
+  const openMainNavbarMenu = useMenuStore(
+    (state: MenuContextProps) => state.openMainNavbarMenu,
+  );
+
+  const toggleMenuHandlers = useMenuStore((state: MenuContextProps) => ({
+    openNavbarWalletMenu: state.openNavbarWalletMenu,
+    onOpenNavbarWalletMenu: state.onOpenNavbarWalletMenu,
+    openNavbarConnectedMenu: state.openNavbarConnectedMenu,
+    onOpenNavbarConnectedMenu: state.onOpenNavbarConnectedMenu,
+  }));
+
+  const onMenuInit = useMenuStore(
+    (state: MenuContextProps) => state.onMenuInit,
+  );
+
   const { t: translate } = useTranslation();
   const i18Path = 'navbar.';
   const walletManagement = useWallet();
   const { account } = useWallet();
 
-  !account.isActive ?? settings.onWalletDisconnect();
+  !account.isActive ?? onWalletDisconnect();
 
   // return focus to the button when we transitioned from !open -> open
-  const prevOpen = useRef(menu.openMainNavbarMenu);
+  const prevOpen = useRef(openMainNavbarMenu);
   useEffect(() => {
-    if (prevOpen.current === true && menu.openMainNavbarMenu === false) {
-      anchorRef.current!.focus();
+    if (prevOpen.current === true && openMainNavbarMenu === false) {
+      anchorRef!.current.focus();
     }
 
-    prevOpen.current = menu.openMainNavbarMenu;
-  }, [menu.openMainNavbarMenu]);
+    prevOpen.current = openMainNavbarMenu;
+  }, [openMainNavbarMenu]);
 
-  useEffect(() => {
-    menu.onMenuInit(anchorRef);
+  useLayoutEffect(() => {
```

Comment:
Does it perform better with useLayoutEffect?

Reply:
Thanks for asking and please confirm my solution or not: For the Popper of that NavbarMenu a Reference is needed. Many times over, I ran into this issue that the Reference passed to the SettingsStore/Context (before) was undefined.

With useLayoutEffect I want to access the latest state of the DOM. So to make sure that the component, serving as ref, is already in place and can be passed to that onMenuInit.